test/e2e/operator_test.go

```diff
@@ -421,6 +420,30 @@ func componentRefEventuallyExists(w watch.Interface, exists bool, ref *corev1.Ob
 	}))
 }
 
+func multipleComponentRefsEventuallyExists(w watch.Interface, exists bool, refs []*corev1.ObjectReference) {
+	deadline, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+
+	awaitPredicates(deadline, w, operatorPredicate(func(op *operatorsv1.Operator) bool {
+		if op.Status.Components == nil {
+			return false
+		}
+		for _, r := range op.Status.Components.Refs {
+			for i := 0; i < len(refs); i++ {
+				if refs[i] != nil && r.APIVersion == refs[i].APIVersion && r.Kind == refs[i].Kind && r.Namespace == refs[i].Namespace && r.Name == refs[i].Name {
+					refs[i] = nil
+				}
+			}
+		}
+		for _, ref := range refs {
+			if ref != nil {
+				return !exists
+			}
+		}
+		return exists
```

Comment:
I think mutating the status could be dangerous. How about something like this?

```suggestion
        foundCount := 0
	awaitPredicates(deadline, w, operatorPredicate(func(op *operatorsv1.Operator) bool {
		if op.Status.Components == nil {
			return false
		}
		for _, r := range op.Status.Components.Refs {
			for i := 0; i < len(refs); i++ {
				if refs[i] != nil && r.APIVersion == refs[i].APIVersion && r.Kind == refs[i].Kind && r.Namespace == refs[i].Namespace && r.Name == refs[i].Name {
					foundCount++
				}
			}
		}
                return foundCount == len(refs)
```